src/Analysis/Engine/Impl/Interpreter/Ast/AstPythonInterpreter.cs

```diff
@@ -274,6 +274,7 @@ internal class AstPythonInterpreter : IPythonInterpreter2, IModuleContext, ICanF
             }
 
             _analyzer = state;
+            RemoveImportedModules();
```

Comment:
Do we need a lock here?

Reply:
Code locks on `_userSearchPathImported` operations

Reply:
I am resolving it on another work with importing/search paths